java/src/org/openqa/selenium/grid/web/GridUiRoute.java

```diff
@@ -65,4 +77,34 @@ public boolean matches(HttpRequest req) {
   public HttpResponse execute(HttpRequest req) {
     return routes.execute(req);
   }
+
+  private static Routable[] uiRoutes(String prefix, Supplier<HttpHandler> handler) {
+    Routable alwaysAdd = Route.prefix("/ui").to(Route.matching(req -> true).to(handler));
+    if (prefix.isEmpty()) {
+      return new Routable[]{alwaysAdd};
+    }
+    return new Routable[]{
+      alwaysAdd,
+      Route.prefix(prefix + "/ui").to(Route.matching(req -> true).to(handler))
+    };
+  }
+
+  private static Routable[] consoleRoutes(String prefix, Supplier<HttpHandler> handler) {
+    Routable alwaysAdd = get("/grid/console").to(handler);
+    if (prefix.isEmpty()) {
+      return new Routable[]{alwaysAdd};
+    }
+    return new Routable[]{
+      alwaysAdd,
+      get(prefix + "/grid/console").to(handler)
+    };
+  }
+
+  private static Routable[] redirectRoutes(String prefix, Supplier<HttpHandler> handler ) {
+    if (prefix.isEmpty()) {
+      return new Routable[]{};
+    }
+    prefix = prefix.endsWith("/") ? prefix.substring(0, prefix.length() -1) : prefix;
+    return new Routable[]{get(prefix).to(handler)};
+  }
```

Comment:
This could be a single method by adding a 3rd parameter, which is the route itself, right?

Reply:
But aren't these three methods doing almost the same? Just with different values for the Route?

Reply:
* `uiRoutes()` works with `Route.prefix()`
* `consoleRoutes()` works with `Route.get()`
* `redirectRoutes()` is applicable only if there was a prefix that was specified.

Unifying them would basically mean that we create one method which internally does all these 3 activities. It would be the same as having 3 explicit methods which explicitly clarify their intent.

Reply:
Please check now. I have refactored it and simplified the logic a bit more.